src/poly/dcrt/sampler/trapdoor/mod.rs

```diff
@@ -83,9 +83,13 @@ impl DCRTTrapdoor {
         };
         debug_mem("p2z_vec generated");
         // create a matrix of d*k x d ring elements in coefficient representation
-        let p2_vecs = parallel_iter!(0..d)
-            .map(|i| split_int64_vec_to_elems(&p2z_vec.slice(0, n * dk, i, i + 1), params))
-            .collect::<Vec<_>>();
+        let mut p2_vecs = Vec::with_capacity(d);
```

Comment:
requires to be sequencial due to heavy load on I/O call on here `split_int64_vec_to_elems` if parallell